nss/aad_auth/main.go

```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+
+	"github.com/ubuntu/aad-auth/internal/logger"
+	"github.com/ubuntu/aad-auth/internal/nss"
+)
+
+func main() {
+	flag.Parse()
+
+	if len(flag.Args()) < 2 {
+		log.Fatal("Not enough arguments.")
+	}
+
+	switch flag.Arg(0) {
+	case "getent":
+		ctx := nss.CtxWithSyslogLogger(context.Background())
+		defer logger.CloseLoggerFromContext(ctx)
+
+		db, key := flag.Arg(1), flag.Arg(2)
+		entries, err := GetEnt(ctx, db, key)
+
+		fmt.Print(fmtOutput(ctx, entries, err))
+	}
+}
+
+func fmtOutput(ctx context.Context, entries []string, err error) string {
+	var out string
+
+	status, errno := 1, 0
```

Comment:
Don’t hardcode 1 please, what is 1? how would I know in like 2 years?

It’s better to import the const from NSS to map it to `NSS_STATUS_SUCCESS`.

Actually errToCStatus handles the nil return values and will map it to NSS_STATUS_SUCCESS, 0, so always call it on err, even if nil?